tests/MongoDB.Driver.Tests/Specifications/server-selection/ServerSelectionTests.cs

```diff
@@ -0,0 +1,93 @@
+/* Copyright 2019-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using System.Threading;
+using FluentAssertions;
+using MongoDB.Bson;
+using MongoDB.Bson.TestHelpers.XunitExtensions;
+using MongoDB.Driver.Core;
+using MongoDB.Driver.Core.Clusters;
+using MongoDB.Driver.Core.Events;
+using MongoDB.Driver.Core.Misc;
+using MongoDB.Driver.Core.TestHelpers.XunitExtensions;
+using MongoDB.Driver.TestHelpers;
+using Xunit;
+
+namespace MongoDB.Driver.Tests.Specifications.server_selection
+{
+    public class ServerSelectionTest
+    {
+        [Theory]
+        [ParameterAttributeData]
+        public void ReadPreference_should_not_be_sent_to_standalone_server(
+            [Values(false, true)] bool async)
+        {
+            var eventCapturer = new EventCapturer().Capture<CommandStartedEvent>(e => e.CommandName.Equals("find") || e.CommandName.Equals("$query"));
+            using (var subject = CreateDisposableClient(eventCapturer, ReadPreference.PrimaryPreferred))
```

Comment:
It's debatable that `MongoClient` is the subject under test here.

Aren't we testing the changes to `CommandUsingCommandMMessageWireProtocol`?

Reply:
Yes, that's the file where the changes were made. Does that mean I shouldn't be `using` a disposable client? Or is the name `subject` poorly chosen on my part?

Reply:
Hm, how about `client` as a variable name?

Reply:
Yes `client` is a good variable name.

I don't think there is any variable in this test that can be named `subject`. This is a very indirect test in which the actual subject under test (the CommandUsingCommandMessageWireProtocol) never actually appears in the test code.

Reply:
Done